src/onnx/parse_shape.cpp

```diff
@@ -43,13 +45,48 @@ struct parse_shape : op_parser<parse_shape>
     {
         if(args.size() != 1)
             MIGRAPHX_THROW("Shape: operator should have 1 operand");
-        std::vector<std::size_t> arg_shape = args[0]->get_shape().lens();
-        std::vector<int64_t> vec_shape(arg_shape.size());
-        migraphx::shape s(migraphx::shape::int64_type, {arg_shape.size()});
-        std::transform(arg_shape.begin(), arg_shape.end(), vec_shape.begin(), [](auto i) {
-            return int64_t(i);
-        });
-        return info.add_literal(migraphx::literal{s, vec_shape});
+        auto input_shape = args[0]->get_shape();
+        int input_ndim = input_shape.ndim();
+        auto is_ind_out_of_bounds = [&](auto ind){ return (ind > (input_ndim)) or (ind < (-1 * input_ndim)); };
+        std::size_t start = 0;
+        std::size_t end = input_shape.ndim();
```

Comment:
If you are going to use `input_shape.ndim()` directly then you should remove the `input_ndim` variable.